### PR TITLE
Feautre: Init command. Change bin/cleaver & wizard

### PR DESCRIPTION
**Pull request by sudodoki**
Instead of #70.

### CODE REVIEW

**Comment by sudodoki** [comment]
Also need to add all same `.option` from main command & update readme/wiki.

**Comment by jdan** [comment]
I'll play around with this tomorrow and give you feedback :)

File: lib/wizard.js [hunk 1]
```diff
@@ -0,0 +1,119 @@
...
+    }];
+
+  prompt.start();
+  // skip the stuff!
+  if (options.y) {
+    prompt.override = {
+      title: probableTitle,
+      input: _s.slugify(probableTitle) + '.md',
+      output: 'index.html',
+      style: '   '
```
Comment (sudodoki) [inline lib/wizard.js]: If you leave this as just empty string, it will still ask for it in the prompt.

File: lib/wizard.js [hunk 2]
```diff
@@ -0,0 +1,119 @@
...
+        deferred.resolve(result);
+      }
+    });
+  }
+  return deferred.promise;
+}
+
+function writeToFile(result) {
+  var presentation = _s.trim(result.input);
+  var message = "author:\n  name: Your Name\n  email: you@example.com\n  twitter: @username\n  url: http:\/\/yourwebsite.com\n";
```
Comment (sudodoki) [inline lib/wizard.js]: could have putten this into mustache template

File: lib/wizard.js [hunk 3]
```diff
@@ -0,0 +1,119 @@
...
+var fs = require('fs');
+var Q = require('q');
+var prompt = require('prompt');
+var merge = require('merge');
+module.exports = {
+  init: function (options, shouldDoOverrides) {
+    options.y = !!shouldDoOverrides;
+    prompt.message = '>>';
+    prompt.delimiter = ':'.green;
+    console.log("This utility will walk you through creating your very own presentation file.");
```
Comment (jdan) [inline lib/wizard.js]: I don't think these log statements are necessary

File: lib/wizard.js [hunk 4]
```diff
@@ -0,0 +1,119 @@
...
+    console.log("Press ^C at any time to quit.");
+    getPrompt(options).then(checkForFile).then(writeToFile).catch(console.log);
+  }
+};
+
+function getPrompt(options) {
+  var deferred = Q.defer();
+  var probableTitle = options.title || _s.capitalize(process.cwd().split('/').slice(-1)[0]);
```
Comment (jdan) [inline lib/wizard.js]: Leave a comment explaining what this does if you don't mind!

File: lib/wizard.js [hunk 5]
```diff
@@ -0,0 +1,119 @@
...
+  var deferred = Q.defer();
+  var inc = result.input;
+  var confirmation = {
+      name: 'confirm',
+      pattern: /^(y|yes|no|n)$/i,                  // Regular expression that input must be valid against.
+      message: 'Y/Yes/No/N only',
+      description: 'File '+inc.green+' already exist. Overwrite?'.grey,
+      default: 'Yes',
+      required: true
+    };
```
Comment (jdan) [inline lib/wizard.js]: Style nit: decrease the indentation of these lines, and put a space around `if {...}` statements

File: lib/wizard.js (hunk 2 above)
Comment (jdan) [inline lib/wizard.js]: At the very least let's split this onto a few lines :)

File: lib/wizard.js [hunk 6]
```diff
@@ -0,0 +1,119 @@
...
+function writeToFile(result) {
+  var presentation = _s.trim(result.input);
+  var message = "author:\n  name: Your Name\n  email: you@example.com\n  twitter: @username\n  url: http:\/\/yourwebsite.com\n";
+  message += ['title', 'theme', 'style', 'output', 'controls', 'progress', 'encoding',
+    'template', 'layout'].filter(function (key) {
+      return _s.trim(result[key]);
+    }).map(function (key) {
+      return key + ": " + result[key];
+    }).join('\n');
+  console.log('Inited empty presentation at '+ presentation.green +' containing:');
```
Comment (jdan) [inline lib/wizard.js]: No need for the "containing" bit, and let's change this to:

```
Presentation created at _____
```

File: lib/wizard.js [hunk 7]
```diff
@@ -0,0 +1,119 @@
...
+  console.log(message);
+  fs.writeFile(presentation, message, function (err) {
+    if (err) { return console.log('Oh no, sorry, there was an ',err.message,' error'); }
+    console.log('Rise and shine! Be sure to run ' + ('cleaver watch ' + presentation).green + ' to start hacking rihgt away');
```
Comment (jdan) [inline lib/wizard.js]: Cute, but a little noisy :)

File: lib/wizard.js [hunk 8]
```diff
@@ -0,0 +1,119 @@
...
+      },
+      {
+        name: 'style',
+        description: 'Presentation style. Write theme name of path/to/file.css',
+        type: 'string',
+        default: options.style || '',
+        required: false
+      }
+    ];
+    prompt.get(promptSeries,function (err, result) {
```
Comment (jdan) [inline lib/wizard.js]: space after "," please!

File: lib/wizard.js [hunk 9]
```diff
@@ -0,0 +1,119 @@
...
+      style: '   '
+    };
+  }
+  prompt.get(titlePrompt, function (err, res) {
+    var title = res.title;
+    if (err) { return deferred.reject(err); }
+    var promptSeries = [
+      {
+        name: 'input',
+        description: 'Input markdown file',
```
Comment (jdan) [inline lib/wizard.js]: Let's just go ahead and generate this for them automatically, based on the title. If they want to move it they can.

File: lib/wizard.js [hunk 10]
```diff
@@ -0,0 +1,119 @@
...
+        type: 'string',
+        default: _s.slugify(title) + '.md',
+        required: true
+      },
+      {
+        name: 'output',
+        description: "Presentation output filename",
```
Comment (jdan) [inline lib/wizard.js]: Let's just do the above option, but replace `.md` with `.html`. I don't think we need to prompt for input or output, what do you think?

**Comment by jdan** [comment]
Heya, this is super awesome - thanks for hacking on cleaver again :)

Sorry for leaving so many comments, mostly style things, but above all I don't know how I feel about asking the user for this many things. The fact that we *allow* them to skip it is a little worrying, cleaver is much simpler than that.

I think we can get away with just asking for a title and stylesheet, to be completely honest. Am I trimming too much out here? What are your thoughts?

**Comment by jdan** [comment]
The reason I say this is because I sort of wrote this functionality into [tinman](http://jdan.github.io/tinman/the-shiniest-blog-engine-in-oz/), and I don't think cleaver is much more complex.

**Comment by firstdoit** [comment]
Closing this for now, feel free to rebase for master and re-open the discussion! As for features, I agree: the less the better.

**Pull request closed without merging**